File: rc-application/src/main/java/com/transkript/reportcard/business/service/SubjectService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.SubjectDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.Subject;

import java.util.List;

public interface SubjectService {

    List<SubjectDto> getSubjects();

    SubjectDto getSubject(Long id);

    EntityResponse addSubject(SubjectDto subjectDto);

    EntityResponse updateSubject(Long id, SubjectDto subjectDto);

    String deleteSubject(Long id);

    Subject getSubjectEntity(Long subjectId);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/interf/SubjectService.kt
package com.transkript.reportcard.business.service.interf

import com.transkript.reportcard.api.dto.SubjectDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.Subject

interface SubjectService {
    val subjects: List<SubjectDto?>?
    fun getSubject(id: Long?): SubjectDto?
    fun addSubject(subjectDto: SubjectDto?): EntityResponse?
    fun updateSubject(id: Long?, subjectDto: SubjectDto?): EntityResponse?
    fun deleteSubject(id: Long?): String?
    fun getSubjectEntity(subjectId: Long?): Subject?
}
